utils/zerovec/src/ule/plain.rs

```diff
@@ -15,53 +15,51 @@ use core::num::{NonZeroI8, NonZeroU8};
 #[allow(clippy::exhaustive_structs)] // newtype
 pub struct RawBytesULE<const N: usize>(pub [u8; N]);
 
-macro_rules! impl_byte_slice_size {
-    ($size:literal) => {
-        impl From<[u8; $size]> for RawBytesULE<$size> {
-            #[inline]
-            fn from(le_bytes: [u8; $size]) -> Self {
-                Self(le_bytes)
-            }
-        }
-        impl RawBytesULE<$size> {
-            #[inline]
-            pub fn as_bytes(&self) -> &[u8] {
-                &self.0
-            }
-        }
-        // Safety (based on the safety checklist on the ULE trait):
-        //  1. RawBytesULE does not include any uninitialized or padding bytes.
-        //     (achieved by `#[repr(transparent)]` on a type that satisfies this invariant)
-        //  2. RawBytesULE is aligned to 1 byte.
-        //     (achieved by `#[repr(transparent)]` on a type that satisfies this invariant)
-        //  3. The impl of validate_byte_slice() returns an error if any byte is not valid (never).
-        //  4. The impl of validate_byte_slice() returns an error if there are leftover bytes.
-        //  5. The other ULE methods use the default impl.
-        //  6. RawBytesULE byte equality is semantic equality
-        unsafe impl ULE for RawBytesULE<$size> {
-            #[inline]
-            fn validate_byte_slice(bytes: &[u8]) -> Result<(), ZeroVecError> {
-                if bytes.len() % $size == 0 {
-                    // Safe because Self is transparent over [u8; $size]
-                    Ok(())
-                } else {
-                    Err(ZeroVecError::length::<Self>(bytes.len()))
-                }
-            }
-        }
+impl<const N: usize> RawBytesULE<N> {
+    #[inline]
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
 
-        impl RawBytesULE<$size> {
-            #[inline]
-            pub fn from_byte_slice_unchecked_mut(bytes: &mut [u8]) -> &mut [Self] {
-                let data = bytes.as_mut_ptr();
-                let len = bytes.len() / $size;
-                // Safe because Self is transparent over [u8; $size]
-                unsafe { core::slice::from_raw_parts_mut(data as *mut Self, len) }
-            }
+    #[inline]
+    pub fn from_byte_slice_unchecked_mut(bytes: &mut [u8]) -> &mut [Self] {
+        let data = bytes.as_mut_ptr();
+        let len = bytes.len() / N;
+        // Safe because Self is transparent over [u8; $size]
```

Comment:
Are you sure? That would change the public interface, since unsizing goes from `[u8; N]`  to `[u8]`, not the other way around. It would change quite a few things, e.g.: https://github.com/unicode-org/icu4x/blob/ce9d4817311ba4e848cdd129e14aeff95937b320/utils/zerovec/src/varzerovec/components.rs#L82

Reply:
Thanks, completely missed that